armi/reactor/converters/uniformMesh.py

```diff
@@ -85,6 +86,321 @@ def converterFactory(globalFluxOptions):
+            list(self._commonMesh) + materialBottoms,
+            self.minimumMeshSize,
+            materialBottoms,
+            preference="bottom",
+        )
+        # combine material top boundaries with full mesh using top preference
+        meshWithTops = self._filterMesh(
+            list(self._commonMesh) + materialTops,
+            self.minimumMeshSize,
+            materialTops,
+            preference="top",
+        )
+        # combine all mesh points using all material boundaries as anchors with top preference
+        combinedMesh = self._filterMesh(
+            list(set(meshWithBottoms + meshWithTops)),
+            self.minimumMeshSize,
+            materialAnchors,
+            preference="top",
+        )
+
+        self._commonMesh = numpy.array(combinedMesh)
+
+    def _filterMesh(
+        self, meshList, minimumMeshSize, anchorPoints, preference="bottom", warn=False
+    ):
+        """
+        Check for mesh violating the minimum mesh size and remove them if necessary
+
+        Parameters
+        ----------
+        meshList : list of float, required
+            List of mesh points to be filtered by minimum mesh size
+        minimumMeshSize : float, required
+            Minimum allowed separation between axial mesh points in cm
+        anchorPoints : list of float, required
+            These mesh points will not be removed. Note that the anchor points must be separated by
+            at least the ``minimumMeshSize``.
+        preference : str, optional
+            When neither mesh point is in the list of ``anchorPoints``, which mesh point is given preference
+            ("bottom" or "top")
+        warn : bool, optional
+            Whether to log a warning when a mesh is removed. This is true if a
+            control material boundary is removed, but otherwise it is false.
+        """
+        if preference == "bottom":
+            meshList = sorted(list(set(meshList)))
+        elif preference == "top":
+            meshList = sorted(list(set(meshList)), reverse=True)
+
+        while True:
+            for i in range(len(meshList) - 1):
+                difference = abs(meshList[i + 1] - meshList[i])
+                if difference < minimumMeshSize:
+                    removeIndex = self._determineIndexToRemove(
+                        i, i + 1, meshList[i : i + 2], anchorPoints
+                    )
+                    if meshList[i] in anchorPoints and meshList[i + 1] in anchorPoints:
+                        errorMsg = (
+                            "Attempting to remove two anchor points!\n"
+                            "The uniform mesh minimum size for decusping is smaller than the "
+                            "gap between anchor points:\n"
+                            f"{meshList[i]}, {meshList[i+1]}, gap = {abs(meshList[i]-meshList[i+1])}"
+                        )
+                        runLog.error(errorMsg)
+                        raise ValueError(errorMsg)
```

Comment:
As is common around these parts, you added this exception, but it is not covered by unit tests.

Reply:
I added a unit test to cover this error